crates/router/src/core/payments.rs

```diff
@@ -1852,6 +1818,90 @@ where
     Ok((router_data, merchant_connector_account))
 }
 
+pub async fn add_decrypted_payment_method_token<F, Req, D>(
+    tokenization_action: TokenizationAction,
+    payment_data: &D,
+    mut router_data: RouterData<F, Req, router_types::PaymentsResponseData>,
```

Comment:
Is it possible to refactor the function so that instead of taking `mut router_data`, it returns only the `PaymentMethodToken`. Then, you can assign it to `router_data.payment_method_token` as needed.

Something like
```
router_data.payment_method_token = add_decrypted_payment_method_token(tokenization_action.clone(), payment_data)
    .await?;

```